docker_images/sklearn/tests/test_api_tabular_classification.py

```diff
@@ -67,6 +67,18 @@ def test_success_code(self):
         assert type(content) == list
         assert len(content) == expected_output_len
 
+    def test_wrong_sklearn_version(self):
+        self._check_requirement(self.case_data["old_sklearn"])
+
+        data = self.data
+        with TestClient(self.app) as client:
+            response = client.post("/", json={"inputs": data})
+
+        assert response.status_code == 200
+        content = json.loads(response.content)
+        # TODO: fix this here.
+        assert content == {}
```

Comment:
I've got an issue here. This is the part of the code in `route.py` which attaches warnings to the response:

``` python
    try:
        outputs = pipe(inputs)
        task = os.getenv("TASK")
        metrics = get_metric(inputs, task, pipe)
    except (AssertionError, ValueError) as e:
        outputs = {"error": str(e)}
        status_code = 400
    except Exception as e:
        outputs = {"error": "unknown error"}
        status_code = 500
        logger.error(f"There was an inference error: {e}")

    if warnings and isinstance(outputs, dict):
        outputs["warnings"] = list(sorted(warnings))
```

The issue is they are only attached if there is at least one exception AND some warnings, otherwise warnings are all ignored.

Here's the question:

Should we change `route.py` and if there are only warnings, return a result which includes the output and the warnings, or should my code here raise an error, include the output in the error message for the user to see, and also raise those warnings?

cc @Narsil

Reply:
Went the latter route. Not touching `routes.py`.